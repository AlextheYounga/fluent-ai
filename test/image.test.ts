import { test, expect } from "bun:test";
import { openai, fal, requestObject, load } from "../src";

const jobs = [
  fal({ apiKey: "<key>" }).image("fal-ai/flux/dev"),
  openai({ apiKey: "<key>" }).image("dall-e-2"),
];

test("image", async () => {
  for (const job of jobs) {
    expect(
      await requestObject(
        job.n(3).prompt("A cute baby sea otter").makeRequest()
      )
    ).toMatchSnapshot();
  }
});

test("dump", () => {
  for (const job of jobs) {
    expect(job.dump()).toMatchSnapshot();
  }
});

test("load", async () => {
  for (const job of jobs) {
    const req1 = await requestObject(load(job.dump()).makeRequest!());
    const req2 = await requestObject(job.makeRequest());
    expect(req1).toEqual(req2);
  }
});
