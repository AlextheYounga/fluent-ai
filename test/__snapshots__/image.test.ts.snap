// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`image 1`] = `
{
  "body": "{"prompt":"A cute baby sea otter","num_images":3}",
  "headers": {
    "authorization": "Key <key>",
    "content-type": "application/json",
  },
  "method": "POST",
  "url": "https://queue.fal.run/fal-ai/flux/dev",
}
`;

exports[`image 2`] = `
{
  "body": "{"prompt":"A cute baby sea otter","model":"dall-e-2","n":3}",
  "headers": {
    "authorization": "Bearer <key>",
    "content-type": "application/json",
  },
  "method": "POST",
  "url": "https://api.openai.com/v1/image/generations",
}
`;

exports[`dump 1`] = `
{
  "image": {
    "model": "fal-ai/flux/dev",
    "params": {
      "n": 3,
      "prompt": "A cute baby sea otter",
    },
  },
  "options": {
    "apiKey": "<key>",
  },
  "provider": "fal",
}
`;

exports[`dump 2`] = `
{
  "image": {
    "model": "dall-e-2",
    "params": {
      "n": 3,
      "prompt": "A cute baby sea otter",
    },
  },
  "options": {
    "apiKey": "<key>",
  },
  "provider": "openai",
}
`;
